src/main/java/com/marketdata/MarketSimulatorBackend/service/AlphaVantageProvider.java

```diff
@@ -0,0 +1,49 @@
+package com.marketdata.MarketSimulatorBackend.service;
+
+import com.marketdata.MarketSimulatorBackend.model.StockPrice;
+import org.json.JSONObject;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class AlphaVantageProvider implements StockServiceProvider {
```

Comment:
I suggest using `public class AlphaVantageStockProvider implements StockProvider` and `public class MockStockProvider implements StockProvider`.